src/Raven.Server/Documents/Queries/Parser/QueryParser.cs

```diff
@@ -391,10 +391,7 @@ private bool GraphAlias(GraphQuery gq,bool isEdge, StringSegment implicitPrefix,
                 {
                     if (isEdge)
                     {
-                        if (Scanner.TryPeek(']'))
-                        {
-                            throw new InvalidQueryException($"Expected to find a field identifier after '{Scanner.Token}', but found ']'. Perhaps this a typo?",Scanner.Input);
-                        }
+                        ThrowIfUnexpectedClosingBracket();
```

Comment:
I've thought about extracting just the exception to `ThrowInvalidQueryException(string message)` and use it in all places where you are throwing the ex (we are doing similar thing with `ThrowParseException`

Reply:
We could have a single function that receives exception message string, but I think that we could postpone it to later refactorings and we could merge this PR, because all significant preformance related issues we addressed, and I'd prefer avoid additional context switches.